### PR TITLE
Keep track of originally programmed basal rate

### PR DESCRIPTION
**Pull request by ps2**
This is based on pod-alarms, so I created the PR against it, just for clarity during review; will rebase against dev when pod-alarms lands.

https://tidepool.atlassian.net/browse/LOOP-899

This also brings in the Dash SDK release-1.0.7-build14, which has fixes for the hang that has been observed.

### CODE REVIEW

File: DashKit/DashPumpManager.swift [hunk 1]
```diff
@@ -535,6 +535,9 @@ public class DashPumpManager: PumpManager {
                 switch result {
                 case .success(let podStatus):
                     self.mutateState({ (state) in
+                        if let finishedBolus = state.unfinalizedBolus {
+                            state.finishedDoses.append(finishedBolus)
+                        }
```
Comment (nhamming) [inline DashKit/DashPumpManager.swift:540]: not sure why an `unfinalizedBolus` is assigned to a `finishedBolus`?

**Review by nhamming — APPROVED** [review]
Just a clarification question.

**Review by darinkrauss — APPROVED** [review]
LGTM 🚀

File: DashKit/DashPumpManager.swift (hunk 1 above)
Comment (ps2) [reply DashKit/DashPumpManager.swift:540]: A dose can be finished in the sense that delivery is finished, but not finalized (i.e. not finished with the events that process it to an unmutable state in persistent storage).

**Pull request merged**